cal/cal/models.py

```diff
@@ -650,6 +684,41 @@ def query(self, calendar_ids=None, start=None, end=None):
 
         return events_qs.order_by('start')
 
+    def queryWeek(self, calendar_ids=None, start=None, end=None):
+        """
+        Returns a QuerySet of events matching this Tag.
+        Does not truncate at the edges.
+        """
+        calendars = self.user.profile.get_calendars_for_calendarids(calendar_ids)
+
+        keywords = self.keywords.split(',')
+        if not keywords:
+            raise InvalidParameterException("No keywords defined for this tag")
+
+        querysets = [
+                GEvent.objects
+                .filter(calendar=calendar, name__icontains=keyword)
+                .exclude(all_day_event=True)
+                for keyword in keywords
+                for calendar in calendars
+                ]
+
+        # Union over the querysets
+        events_qs = reduce(lambda qs1, qs2: qs1 | qs2, querysets)
+
+        queryset = []
+        events = events_qs.order_by('start')
```

Comment:
this looks like a copy and paste of the code earlier, perhaps we could generalize this code and have each class pull from that generalized code instead?